### DISCUSSION TITLE
欢迎来到铁蛋ROS 2讨论区 - Welcome to cyberdog_ros2 Discussions!

### DISCUSSION THREAD

**Discussion by homalozoa** [MiRoboticsLab/cyberdog_ros2#4] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

上边这一堆是默认的，看着办就行[手动狗头]

欢迎大家踊跃讨论，不限制题材。

**Comment by yulong88888** [comment]
🐂  🍺  ～

**Comment by carlzhusz** [comment]
请问下还有哪个论坛讨论铁蛋开发的？

**Reply by homalozoa** [reply]
你在铁蛋的论坛讨论还有哪个论坛能讨论铁蛋？

**Comment by Roger200** [comment]
￼￼Hi, I'm new and my name is Roger, I have questions below regarding Cyberdog's functionality, _

1. 请问Cyberdog,可以換语言吗？

Cyberdog is not responding to my voice commands such as shake hands etc... is there something that I need to activate ? 

2.还有他不听我的语音指令，为什么呢？

Thank you

**Reply by mi-CyberDog** [reply]
1.Voice currently only supports Chinese

**Reply by mi-CyberDog** [reply]
2.Please try to clear the APP data and log in to the Xiaomi account again

**Reply by Roger200** [reply]
1. How can I make Cyberdog back flip? Like in the video, what command is it? Thanks 

2. Also what mouse can I plug in to control? Usb C ?  I'm scratching my head 

3. And where is the map saved in the app within the phone? So I can backup 

4. I'm connected to Cyberdog jetson , what is User id mi's password ? And mi is not a super user , what is the super user ID and password ???

**Comment by Roger200** [comment]
請問mi 用戶密碼是什么呢？还有請告訴￼super user 的ID 和￼￼￼密碼？


1. How can I make Cyberdog back flip? Like in the video, what command is it? Thanks

2. where is the map saved in the app within the phone? So I can backup

It is really lack of Support here !!!

3. I connected to the Cyberdog's jetson , what is the SSH password to connect ???? For user mi , 

4. Also what is the super user ID and password ??? 

Please reply ASAP!!!!

**Reply by XUANJIE6** [reply]
ssh mi@192.168.1.222
192.168.1.222 这需要你用显示屏连在狗的hdmi接口上，输入命令查询自己的ip地址。密码是123
希望能帮到您。如有问题可以在这问。

**Comment by Roger200** [comment]
好的谢，

还有我有以下的题，


  1.  在（https://youtu.be/frkpEbtJytY）1:10 分，铁蛋可以反筋斗，但是我命令唇疃罚�]有反槭裁茨兀
  2.  用你的铁蛋 app做完地D之后，我在那e可以做个地D浞菽兀
  3.  在铁蛋里，我想M去/ (root), 但是mi用户没有嘞蓿媚且桓鲇没в嘞弈兀棵艽a是？
  4.  还有在里面那里可以找到Ros 或ros2 的工作空g呢？比如： catkin_ws 或ros2_ws ... 因槲铱床坏
  5.  你有什么n程我更加可以明白铁蛋的Y和背後的\作吗？

期待你的回覆

________________________________

**Reply by XUANJIE6** [reply]
1. 我在国内用不了youtu，而且翻筋斗这个功能在我的铁胆上也没有。可能是发行版的铁蛋限制了电机的性能，从而无法实现。
2. 你的翻译都是乱码。翻译的很差。建议你用英语直接问
3. 由于翻译问题无法回答其他问题

**Comment by Roger200** [comment]
Ok thanks for your reply, I will use English then if you are ok with it , you can reply in Chinese if you like as I can read Chinese 你可以用中文回答我的，我会看中文


  1.   ok.... that's very misleading then, because I thought Cyberdog can do back flips ..... that's a big selling point in your demonstration videos ... and you telling me there is no way cyber dog can do it ...

     2. Now I connected hdmi to a screen and I want to open some folders like the "root" folder, it doesn't allow me because the account "mi" does not have permissions , what Super User account credentials can I use ? In order to open the "root" folder

3.also inside , I do not see any ROS workspace, can you let me know where I can find the Cyberdog ROS or ROS2 workspace please? Such as catkin_ws / ros2ws etc...

4. I used your Cyberdog app to make a map, please tell me where the new map is stored inside the Android phone? So that I can make a backup of the map

5.do you have any internal courses or materials online you can share with me ? About the structure of Cyberdog and how everything works in the back ground ? Such as topics, nodes , connections, voice recognitions , how it works etc.....

I look forward to your reply thanks 谢谢





Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by XUANJIE6** [reply]
1. 我不是开发者，只是一个购买了铁蛋的消费者，因为受到另一位购买者的帮助，所以会在这回答一些我力所能及的问题。
2. 在你输入的命令前加 sudo 就是管理员权限，比如打开并修改一个需要权限的文件  sudo vim ~/.bashrc  ,按回车键后会提示你输入密码，密码为 123    ，在比如给予USB串口以权限 sudo chmod 777 /dev/ttyUSB0   这些不加sudo都是没法操作的
3. ROS的工作空间是自己建立的，例如：开启一个命令行输入 mkdir -p chengon/src   ，这时候你就会在根目录下有一个chengon文件夹，把你要编译的文件放在src文件下。在chengon的同级目录下打开命令行 输入 colcon build  
4. ROS2 在/opt/  里，输入cd /opt/   再 ls  你就能看见了
5. 这建立好的地图好像是存在铁蛋里面的。由于我技术能力有限，这个问题无法回答
6. 一，在电脑端安装ros2 通讯改方式为cyclonedds,因为foxy版本默认的是fastdds。galactic好像是默认cyclonedds。二，按github上的 CyberDog DDS本地及多播设置 教程进行设置铁蛋的设置。三，在铁蛋端输入env | grep ROS 查看ROS_DOMAIN_ID= 42   ，我是42，所以在电脑端对bashrc文件加入 export  ROS_DOMAIN_ID= 42。基本就能进行通信了，也就是能在你的电脑上查看到铁蛋的话题，节点，什么的。
7. 能力有限，我是个菜鸟。以上都是网名为 海总 以及 NCPA_super  两位大佬手把手教我的，在这里献上诚挚的感谢！！！

**Reply by XUANJIE6** [reply]
铁蛋的确可以后空翻，前提是你要有足够的编程技术对铁蛋进行二次开发。而且我认为铁蛋作为一个开发平台。它的亮点不是后空翻。而是让我们拥有一个低价接触并开发四足机器人的机会，这机会很少，世界上只有不到一千人拥有这难得的机会。这才是铁蛋的亮点。

**Comment by Roger200** [comment]
ok I see, I thought you were officially from Xiaomi Support, I really appreciate you helping me out then thank you. Are you on wechat then? maybe we can share more experiences and ideas together since we have purchased the same Cyberdog, I'm a newbie and learning here and there.


  1.  good idea, I will give sudo a try
  2.  ok I see, so there is a ROS2 workspace  in /opt folder , is this workspace codes exactly the same as the ones in here on the github link below?(if you know)  https://github.com/MiRoboticsLab/cyberdog_ros2
[https://repository-images.githubusercontent.com/406747198/9e2bc987-0981-4cc5-ac44-970c9a21b92a]<https://github.com/MiRoboticsLab/cyberdog_ros2>
GitHub - MiRoboticsLab/cyberdog_ros2: ROS 2 packages for CyberDog<https://github.com/MiRoboticsLab/cyberdog_ros2>
Xiaomi CyberDog ROS 2. English. 简介. 本项目包含小米铁蛋®的ROS 2主要功能包. 基本信息. 铁蛋默认用户是mi, 密码为123; 使用USB线连接Download接口, 可通过ssh ***@***.***连接铁蛋进行内部操作; 软件架构
github.com
  3.  3. So inside this ROS2 workspace within /opt, do you understand the codes for the functions and how it works? (just wondering...) because ROS2 is very new to me, I'm more confident with ROS1

           4. And do you know if Cyberdog's functions and startups, drivers are all activated within the ROS2 workspace in /opt? (or are these functions, drivers, startups launch etc... are activated somewhere else?)


  1.  5. do you know if there is an official Xiaomi support on this Cyberdog? because I try searching on the web and couldn't find any official contact on this robot. It is very mysterious.

           Thank you and look forward to your reply.

________________________________

**Reply by XUANJIE6** [reply]
1. 我对铁蛋的理解程度也只有上次回答你的那么多。中国国内关于ROS2的资料都是从你们国外翻译过来的，而且非常很少。我也在学习ros2，在github里学习。
2. 我认为你问的第三第四个问题可以通过代码的阅读得到答案，虽然我还没开始这项工作。
3. 有一个不太好的消息，铁蛋这个项目似乎结束了。我认为这个项目对于小米公司来说，只是一个测试与积累，并不是一个主要产业。达到他们的预期目标后，这个项目就结束了。

**Comment by Roger200** [comment]
I see, I was watching the youtube demonstration and I thought that the 'backflip' was already a function included in Cyberdog when we purchase, but from what you said, it is not included out of the box and it looks like Xiaomi has kept the function away from consumers ....and for us....., we have to build on top to make it work, I'm wondering if this backflip function would be hidden in github.....,

I guess you are right, the focus is more on the Cyberdog with four legs that we can further develop and build on. Wow, really, so there is only 1000 people in the whole world, how did you know that? I think we are very very lucky then.


________________________________

**Reply by XUANJIE6** [reply]
九千人民币的铁蛋，它的各个零部件采购家加起来都上万了，更不用区计算开发人员的工资和其他开发成本。九千元人民币是雷总和海总想和大家交个朋友给的亏本价格。因为不赚钱，所以他们交了一千个朋友后就停止生产了。。停产了，每后续了。

**Comment by Roger200** [comment]
Well thank you for your respond back here to me, much appreciated!

It is pity that Xiaomi has closed this project so quickly, it is sad to hear that this is only a test. They could of used this Cyberdog as an education course on their Mi academy website or something, very pity! Now can learn nothing out of it ...


________________________________

**Comment by MiniStrikers** [comment]
Hi,

Why is my cyberdog not able to smell anything as shown in several websites videos where it can smell soy sauce, tea, coffee, mixed juice, garlic, onion, orange and coriander as shown here: https://en.xiaomitoday.it/xiaomi-cyberdog-gas-odori.html.

Is it because it hasn't been implemented yet? I am currently using a release package of App Version 1.0.0.46 and Firmware version 1.0.0.94. Pls help. Tks.

**Comment by XUANJIE6** [comment]
https://github.com/MiRoboticsLab/cyberdog_ros2/wiki/%E5%A6%82%E4%BD%95%E7%BA%BF%E5%88%B7%E9%93%81%E8%9B%8B          这个是升级版本的方法。

https://github.com/MiRoboticsLab/cyberdog_ros2/discussions/133  这个是最新版本的下载地址
闻味道这功能我不感兴趣，我也没有尝试过。

**Reply by XUANJIE6** [reply]
http://cdn.cnbj2m.fds.api.mi-img.com/cyberdog-package/build/athena_foxy_2022.01.14_emmc_nvme_V1.0.0.94_release_b1b4a851ca.tgz 这个是最新的。
你能看懂中文最好。因为我不会用英文回答问题。My English is poor

**Reply by MiniStrikers** [reply]
do you have the apk version of it for android?

**Reply by XUANJIE6** [reply]
http://cdn.cnbj1.fds.api.mi-img.com/ota-packages/apk/cyberdog_app.apk

**Reply by MiniStrikers** [reply]
still can't smell though :(

**Reply by Roger200** [reply]
Personally I don't think Cyberdog can actually smell, looking at all the sensors, possibly I guess it may be done by it's front interactive AI camera , what do yous think?

**Reply by MiniStrikers** [reply]
but it doesn't work when I put it in front as well.

**Reply by Roger200** [reply]
You may need to find out the trained models that are inside the dog (I'm frustrated at the lack of documentations overall as well....) 

Also try rqt graph, that may gives glues on what is actually present when Cyberdog initialises

**Reply by XUANJIE6** [reply]
似乎这个功能只是在小米实验室的那条狗上实现了，更新的最新版本说明里没有气味识别功能。

**Reply by Roger200** [reply]
Where did you get those details? Do you have the link to share? How was it done in the lab then, was it via the ai camera ? 

Thanks

**Reply by XUANJIE6** [reply]
https://www.xiaomi.cn/board/27817860   当然是去官网上查看了，这是小米社区/圈子/cyberdog的连接。基本官方的东西基本都在这

**Comment by MAVProxyUser** [comment]
CyberDog Owners are welcome to come discuss dog use here on Slack: https://join.slack.com/t/robotdogs/shared_invite/zt-1fvixx89u-7T79~VxmDYdFSIoTnSagFQ

**Reply by liseri** [reply]
oh, the project is alread alive?

**Reply by MAVProxyUser** [reply]
> oh, the project is alread alive?

The telegram, and Github from "official" support is very dead. Seems the manager has made all the Xiaomi employees tired, and sad with their work. Not sharing open source, and URDF, backflip code, motion controller open source, etc. Community is very eager to be alive, Xiaomi managers are killing the community with lack of support. 

A few dog owners talking together, but "official" after-sales support is very weak.

**Comment by Roger200** [comment]
Exactly and I thought Cyberdog was supposed to be open source after purchase,

where is the open source, and URDF, backflip code, stairs code , motion controller etc...?

it is definitely not what it was advertised before I purchased it.....,


________________________________

**Comment by liseri** [comment]
铁蛋还活着么

**Comment by Roger200** [comment]
公司@b狗的支持很差，所以我建h你L一下unitree狗，你可能玫礁玫氖坩嶂С郑侨绻闶煜ros2，你可以\行python文件它在]有贸淌降那r下移樱Ｄ愫眠\

取得 Android 版 Outlook


________________________________

**Comment by ElaineAintLaine** [comment]
想问一下该如何处理才能取得铁蛋二代的开发权限？

**Reply by liseri** [reply]
铁蛋二代没开源吗？ 我的一代已经落灰好久了；
